ObjectOrientedProgramming/MemoryOwnership/RAIICopySwapIdiom/src/dynamic_array.h

```diff
@@ -0,0 +1,41 @@
+#ifndef CPPBASICS_DYNAMIC_ARRAY_H
+#define CPPBASICS_DYNAMIC_ARRAY_H
+
+#include <iostream>
+
+class dynamic_array {
+private:
+    int *m_data = nullptr;
+    std::size_t m_size = 0;
+
+public:
+    dynamic_array();
+
+    explicit dynamic_array(std::size_t size, int init);
```

Comment:
`explicit` specifier makes sense only for single-argument constructors:
https://en.cppreference.com/w/cpp/language/explicit

But anyway, I would propose to actually remove second argument `init` here, and just don't initialize the memory at all --- the C++ way :)

The reason is that for a proper convenient API we actually need more custom constructors (e.g. a constructor taking `std::initializer_list`). So for simplicity I would propose to leave just one constructor taking size as argument.